### PR TITLE
coordinate successful and failing fetches

### PR DESCRIPTION
**Pull request by stuartnelson3**
## Motivation/summary

make sure that the servers all receive their
requests before confirming that a successful
response from one server cancels the shared
context in the fetch code.

## How to test these changes

non-functional change

## Related issues

closes https://github.com/elastic/apm-server/issues/5836

### CODE REVIEW

**Review by estolfo — APPROVED** [review]
looks good, thanks for doing this

File: sourcemap/fleet_test.go [hunk 1]
```diff
@@ -123,9 +152,21 @@ func TestFailedAndSuccessfulFleetHostsFetch(t *testing.T) {
...
 
-	consumer, err := f.Fetch(context.Background(), name, version, path)
-	assert.NoError(t, err)
-	assert.NotNil(t, consumer)
+	go func() {
+		consumer, err := f.Fetch(context.Background(), name, version, path)
+		resc <- response{consumer, err}
+	}()
+	// Make sure every server has received a request
+	wg.Wait()
```
Comment (marclop) [inline sourcemap/fleet_test.go:160]: Can't we remove this wait group? It seems to me that the channels are already making the test wait until we've received all the signals. What is this wait group adding on top of that?

Comment (stuartnelson3) [reply sourcemap/fleet_test.go:160]: The waitgroup is making sure that each handler has received a request before we attempt to read from any of the channels. Without that, we could potentially read from `errc` and `successc` without the third handler ever being invoked.

Comment (marclop) [reply sourcemap/fleet_test.go:160]: got it, feel free to merge, thanks for the explanation

**Pull request merged**